src/connector/src/sink/kafka.rs

```diff
@@ -395,36 +401,51 @@ impl KafkaSinkWriter {
 
     /// The actual `send_result` function, will be called when the `KafkaSinkWriter` needs to sink
     /// messages
-    async fn send_result<'a, K, P>(&'a self, mut record: FutureRecord<'a, K, P>) -> KafkaResult<()>
+    async fn send_result<'a, K, P>(
+        &'a mut self,
+        mut record: FutureRecord<'a, K, P>,
```

Comment:
Any reason for `record` to be mutable? IIUC, it should be immutable after created.

Reply:
This is due to when the enqueue buffer is full, `send_result_inner` will immediately return `Err(KafkaError, FutureRecord)`, we may reassign the returned record to `record` and retry til the preset limit. Since immutable variable can not be assigned twice, the mutable here is thus needed.
It may be okay to use the same `record`, but I'd prefer the current approach.